changelogs/unreleased/gh-6506-allow-spurious-wakeups-in-wal_write.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/core
+
```

Comment:
While the patch definitely fixes one problem, I wouldn't say it closes the ticket. The ticket was about "dangerous cases" which `fiber_wakeup()` can cause. We still have them in `gc_checkpoint_fiber_f()`, `iproto_do_cfg()` (should be easy to reproduce as well), `box_raft_write()`, `txn_limbo_wait_complete()` and more. I would say to close this ticket we need to drop `fiber_set_cancellable()` completely, it is too dangerous. This commit can be `Part of #6506` but not `Closes`.

Reply:
I've created a separate issue for this: #7166